File: 01.Java-Basic/08.WhileLoop/src/whileLoop/lab/Moving.java
package whileLoop.lab;

import java.util.Scanner;

public class Moving {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int a = Integer.parseInt(scanner.nextLine());
        int b = Integer.parseInt(scanner.nextLine());
        int h = Integer.parseInt(scanner.nextLine());
        int free = a * b * h;

        String input = scanner.nextLine();
        while (!input.equals("Done")) {
            int boxes = Integer.parseInt(input);
            free -= boxes;
            if (free <= 0) {
                break;
            }
            input = scanner.nextLine();
        }

        if (free >= 0) {
            System.out.printf("%d Cubic meters left.\n", free);
        } else {
            System.out.printf("No more free space! You need %d Cubic meters more.", Math.abs(free));
        }
    }
}

File: JavaScript/01.JavaScriptBasic/09While-Loop-Lab/src/moving.js
function canMove(params) {
    let width = Number(params[0]);
    let length = Number(params[1]);
    let height = Number(params[2]);
    let volume = width * length * height;
    let index = 3;
    let data = params[index];
    while (data !== "Done") {
        let boxes = Number(data);
        volume -= boxes;
        if (volume < 0) {
            console.log(`No more free space! You need ${Math.abs(0 - volume)} Cubic meters more.`);
            break;
        }

        index++;
        data = params[index];
    }

    if (volume >= 0) {
        console.log(`${volume} Cubic meters left.`);
    }
}

canMove(["10", 
"10",
"2",
"20",
"20",
"20",
"20",
"122"]);

canMove(["10",
"1",
"2",
"4",
"6",
"Done"]);